File: rlwe-challenges/parsers/python/SecretParser.py
#!/usr/bin/python

# Parses RLWE secret files:
# takes the path to a .secret file and prints the contents.
# See comments in the code for examples of how to access message contents.
#
# You will need the "protobuf" python package to run this code, which you can
# install with "easy_install protobuf"

# You can find more detailed information about parsing messages in python here:
# https://developers.google.com/protocol-buffers/docs/reference/python-generated
import sys
import Challenges_pb2

def parse_secret(secret_path):
  if secret_path is None:
    return
  secret = None
  with open(secret_path) as f:
    secret = Challenges_pb2.Secret()
    secret.ParseFromString(f.read())
  return secret

if __name__ == "__main__":

  if len(sys.argv) != 2:
    print "Usage:", sys.argv[0], "path/to/.secret"
    sys.exit(-1)

  secret = parse_secret(sys.argv[1])

  if secret is None:
    print "Could not parse secret."
    sys.exit(-1)
  else:
    print secret # print all contents
    #print secret.s.xs[0] # it's easy to access any member of the parsed message

  sys.exit(0)

File: rlwe-challenges/parsers/java/SecretParser.java

// This file prints the contents of a secret file.
// The path to a .secret file must be provided.
// See comments in the code for examples of how to access message contents.

// The included jar file was obtained from http://search.maven.org/remotecontent?filepath=com/google/protobuf/protobuf-java/2.6.1/protobuf-java-2.6.1.jar
// > javac -cp protobuf-java-2.6.1.jar crypto/proto/lol/*.java
// > javac -cp .:protobuf-java-2.6.1.jar crypto/proto/RLWE/*.java
// > javac -cp .:protobuf-java-2.6.1.jar crypto/proto/RLWE/challenges/*.java
// > javac -cp .:protobuf-java-2.6.1.jar SecretParser.java
// > java  -cp .:protobuf-java-2.6.1.jar SecretParser path/to/.secret

import crypto.proto.RLWE.challenges.*;
import java.io.FileInputStream;
import java.io.IOException;
import java.io.PrintStream;

class SecretParser {
  public static void main(String[] args) throws Exception {

    // read a secret file
    Challenges.Secret secret =
      Challenges.Secret.parseFrom(new FileInputStream(args[0]));
    System.out.println(secret);
    //System.out.println(secret.getS().getXs(0));
  }
}
